contrib/database/sql/option.go

```diff
@@ -83,3 +86,12 @@ func WithChildSpansOnly() Option {
 		cfg.childSpansOnly = true
 	}
 }
+
+// WithSQLCommentInjection enables injection of tags as sql comments on traced queries.
+// This includes dynamic values like span id, trace id and sampling priority which can make queries
+// unique for some cache implementations. Use WithStaticTagsCommentInjection if this is a concern.
+func WithSQLCommentInjection(mode tracer.SQLCommentInjectionMode) Option {
```

Comment:
I don't hate the idea and that's a small change if you'd like me to make it but it feels a little bit like hiding a problem that should ideally be addressed by bigger changes. 

Taking a step back, the tracer is very protective of data that it uses/propagates while also not allowing much flexibility over injection behavior. Those constraints made it necessary for the sql comment injection to live in the tracer so it could have access to the attributes to inject. But, ideally, this would have lived in some shared sql-related package that could be used by `contrib/database/sql` but also other database packages that can skip the standard `database/sql` compatibility (like [pgx](https://pkg.go.dev/github.com/jackc/pgx#section-readme)). The point here being that I think the sql comment injection mode might not be specific to `database/sql`. Or maybe we expose it that way but then we have to be ready for having copies of this in other contrib wrappers eventually.

Reply:
> I don't hate the idea and that's a small change if you'd like me to make it but it feels a little bit like hiding a problem that should ideally be addressed by bigger changes.

I agree. Good point. Let's keep it as it is. I wasn't too sure either. Seems like the wrong use of type aliases.